pkg/kubelet/dockershim/docker_streaming.go

```diff
@@ -77,17 +81,21 @@ func (r *streamingRuntime) PortForward(podSandboxID string, port int32, stream i
 
 // ExecSync executes a command in the container, and returns the stdout output.
 // If command exits with a non-zero exit code, an error is returned.
-func (ds *dockerService) ExecSync(_ context.Context, req *runtimeapi.ExecSyncRequest) (*runtimeapi.ExecSyncResponse, error) {
+func (ds *dockerService) ExecSync(ctx context.Context, req *runtimeapi.ExecSyncRequest) (*runtimeapi.ExecSyncResponse, error) {
 	timeout := time.Duration(req.Timeout) * time.Second
 	var stdoutBuffer, stderrBuffer bytes.Buffer
-	err := ds.streamingRuntime.exec(req.ContainerId, req.Cmd,
+	err := ds.streamingRuntime.exec(ctx, req.ContainerId, req.Cmd,
 		nil, // in
 		ioutils.WriteCloserWrapper(ioutils.LimitWriter(&stdoutBuffer, maxMsgSize)),
 		ioutils.WriteCloserWrapper(ioutils.LimitWriter(&stderrBuffer, maxMsgSize)),
 		false, // tty
 		nil,   // resize
 		timeout)
 
+	if errors.Is(err, context.DeadlineExceeded) {
+		return nil, status.Errorf(codes.DeadlineExceeded, err.Error())
```

Comment:
remote_runtime.go expects a grpc error with status code `DeadlineExceeded`:

https://github.com/kubernetes/kubernetes/blob/4bb30c3b0e9167eb85b1e3b0c9a3bab3b351a3e2/pkg/kubelet/cri/remote/remote_runtime.go#L394-L397

Reply:
maybe this comment can be added into the code. It is not very logical when you simply looking at this method